twenty-first/src/util_types/merkle_tree.rs

```diff
@@ -179,15 +201,18 @@ impl MerkleTree {
         Ok(nodes)
```

Comment:
After the `MAX_*` constants are gone, this could become something like the following:

```rust
let num_nodes = 2 * num_leafs;
let mut nodes = Vec::new();
nodes
    .try_reserve_exact(num_nodes)
    .map_err(|_| MerkleTreeError::TreeTooHigh)?;
nodes.resize(num_nodes, Digest::default());
nodes[num_leafs..].copy_from_slice(leafs);

Ok(nodes)
```

Overcommitting memory allocators might throw wrenches in gears. ⚙️🔧⚙️